src/olympia/devhub/forms.py

```diff
@@ -665,34 +662,42 @@ def clean(self):
         for locale in name_summary_locales:
             val_len = len(name_values.get(locale, name_default) +
                           summary_values.get(locale, summary_default))
-            if val_len > 70:
-                if locale == default_locale or not self.should_auto_crop:
-                    # if we're not auto-cropping add an error.
+            if val_len > self.MAX_LENGTH:
+                if locale == default_locale:
+                    # only error in default locale.
+                    formatted_message = message.format(
+                        limit_value=self.MAX_LENGTH, show_value=val_len)
                     self.add_error(
                         'name', LocaleErrorMessage(
-                            message=message.format(val_len), locale=locale))
-                else:
+                            message=formatted_message, locale=locale))
+                elif self.should_auto_crop:
                     # otherwise we need to shorten the summary (and or name?)
                     if locale in name_values:
                         # if only default summary need to shorten name instead.
-                        max_name_length = (68 if locale in summary_values
-                                           else 70 - len(summary_default))
+                        max_name_length = (
+                            self.fields['name'].max_length
+                            if locale in summary_values
+                            else self.MAX_LENGTH - len(summary_default))
                         name = name_values[locale][:max_name_length]
                         name_length = len(name)
+                        if 'name' not in self.cleaned_data:
+                            self.cleaned_data['name'] = {}
```

Comment:
This could be shortened a bit by using `self.cleaned_data.setdefault('name', {})`